src/login.js

```diff
@@ -23,23 +24,59 @@ let providers = {
 class Login extends Component {
   constructor(props) {
     super(props);
+    
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+    
+    this.state = {
+      username: '',
+      password: ''
+    }
   }
-
-  renderUserCredentialsEnabled() {
-    if (this.props.userCredentialsEnabled) {
+  
+  handleChange(e) {
+    const target = e.target;
+    const name = target.name;
+    this.setState({
+      [name]: target.value
+    });
+  }
+  
+  handleSubmit(e){
+    e.preventDefault();
+    const options = {
+      uri: this.props.apiAuthorization,
+      qs: {
```

Comment:
According to https://tools.ietf.org/html/rfc6749#section-4.3.2 data should be sent by POST method in `application/x-www-form-urlencoded`. Also scope attribute/property is missing.

Reply:
Base on section-4.3.2, I also removed client_id.